### PR TITLE
post: ng and cli v15.2

### PR DESCRIPTION
**Pull request by cexbrayat**

### CODE REVIEW

File: _posts/2023-02-22-what-is-new-angular-15.2.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+migrating your components/pipes/directives to their standalone versions,
+and removing the obsolete modules of your application 😍.
+
+Sounds interesting? We wrote a guide about it:
+
+👉 [Migrate to standalone components with Angular schematics](/2021/02/15/migrate-an-angular-application-to-standalon/)
+
+## Angular Signals
+
+The Angular team has been working on a different way to handle state and changes in your application for the past year.
```
Comment (jnizet) [inline _posts/2023-02-22-what-is-new-angular-15.2.md:32]: I'd change "state and changes" to "reactivity"

File: _posts/2023-02-22-what-is-new-angular-15.2.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+The first step of the result has been publicly released (even if there is nothing to use yet):
+the [discussion about Angular Signals](https://github.com/angular/angular/discussions/49090).
+
+Signals are a concept that is used in many other frameworks, like SolidJS, Vue, Preact and even the venerable KnockoutJS.
+The idea is to offer a few primitives to define reactive state in your application
+and to allow the framework to only re-render the parts of your application that are impacted by a change.
```
Comment (jnizet) [inline _posts/2023-02-22-what-is-new-angular-15.2.md:38]: I'd rephrase as "and to allow the framework to know which components are impacted by a change, rather than having to detect changes on the whole tree of components."

because Angular *already* re-renders only the parts that are impacted by a change. But it has to detect the changes everywhere.

File: _posts/2023-02-22-what-is-new-angular-15.2.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+This would be a significative change to how Angular works, as it currently relies on zone.js to detect changes and re-render the whole tree of components by default.
```
Comment (jnizet) [inline _posts/2023-02-22-what-is-new-angular-15.2.md:40]: I'd rephrase as "to detect changes in the whole tree of components by default".

File: _posts/2023-02-22-what-is-new-angular-15.2.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+Instead, with signals, the framework would only re-render the components that are impacted by a change.
```
Comment (jnizet) [inline _posts/2023-02-22-what-is-new-angular-15.2.md:41]: > would know which components are impacted, without needing to check all the other ones. 

This also opens the door to zoneless applications, i.e. applications where Angular applications don't need to include Zone.js (which makes them lighter), and don't have to patch all the browser APIs (which makes them start faster).

File: _posts/2023-02-22-what-is-new-angular-15.2.md [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    // define a signal
+    const count = signal(0);
+    // get the value of the signal
+    const value = count();
+    // set the value of the signal
+    count.set(1);
+    // update the value of the signal, based on current value
+    count.update((value) => value + 1);
+    // mutate the value of the signal (handy for objects/arrays)
+    count.mutate((user) => user.name = 'Cédric');
```
Comment (jnizet) [inline _posts/2023-02-22-what-is-new-angular-15.2.md:54]: weird to use the same `count` signal here.

File: _posts/2023-02-22-what-is-new-angular-15.2.md [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+
+Once you have define signals, you define computed values that derive from them:
```
Comment (jnizet) [inline _posts/2023-02-22-what-is-new-angular-15.2.md:56]: define -> defined
you define -> you can define

File: _posts/2023-02-22-what-is-new-angular-15.2.md [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+    count.set(2);
+    console.log(double()); // logs 4
+
+Note that they are lazily computed and only re-computed
+when one of the signals they depend on produces a new value.
+
+Finally, you can use the `effect` function to react to changes in your signals:
+
+    // log the value of the count signal when it changes
+    effect(() => console.log(count()));
```
Comment (jnizet) [inline _posts/2023-02-22-what-is-new-angular-15.2.md:71]: Are you sure about that? See https://github.com/angular/angular/pull/49091#pullrequestreview-1300300096.

File: _posts/2023-02-22-what-is-new-angular-15.2.md [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+
```
Comment (jnizet) [inline _posts/2023-02-22-what-is-new-angular-15.2.md:72]: I would maybe add that, despite looking like a BehaviorSubject, it has some subtle differences, the most important one being that unsubscribing is unnecessary thanks to the usage of weak references.

File: _posts/2023-02-22-what-is-new-angular-15.2.md [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+That's pretty much it for now!
+The next step is to integrate this API with the framework,
+and make it interoperate with RxJS.
+
+In an ideal future,
+we may be able to build a component with fields that are signals and computed values used in the template.
+The framework would then only re-render the component when one of the signals it depends on changes,
```
Comment (jnizet) [inline _posts/2023-02-22-what-is-new-angular-15.2.md:79]: would be notified when an expression has changed thanks to the signals, and would thus only need to re-render the components affected by the change, without having to check for changes on unrelated components, without the need for zone.js

**Review by jnizet — CHANGES_REQUESTED** [review]

File: _posts/2023-02-22-what-is-new-angular-15.2.md (hunk 7 above)
Comment (cexbrayat) [reply _posts/2023-02-22-what-is-new-angular-15.2.md:71]: Yeah, it's not exposed publicly in the current PR, but I suppose there will be something similar in the end as effects/watcher are a necessary part. Maybe this name and others will change as well as the PR is not merged, but it's just to give an idea of it will look like
